plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/ExternalIssue.java

```diff
@@ -53,4 +54,11 @@ public interface ExternalIssue extends IIssue {
    */
   RuleType type();
 
+  /**
+   * Characteristic of the issue according to Clean Code Taxonomy.
+   * @since 9.16
+   */
+  @CheckForNull
```

Comment:
CheckForNull assumes this field is optional. I think we need to figure out if there's a good default value, as we had for types.